### PR TITLE
Implemented Dirichlet BCs with CEED and cleaned up water source updates

### LINKED ISSUE #95: Add CEED implementation to RDySetWaterSource
See [this comment](https://github.com/RDycore/RDycore/pull/94#discussion_r1366175317).

### PR DESCRIPTION
**Pull request by jeff-cohere**
This PR also renames some files and functions to provide slightly shorter and clearer names. It also moves some SWE-related code into the `swe` folder.

We discussed this item in #83, which is more focused on boundary flux accumulation, but related.

Closes #95

### CODE REVIEW

File: src/rdydata.c [hunk 1]
```diff
@@ -23,60 +23,61 @@ static PetscErrorCode CheckBoundaryConditionID(RDy rdy, PetscInt bnd_cond_id) {
...
+    CeedOperatorFieldGetVector(dirichlet_field, &dirichlet_vector);
+    PetscInt num_comp = 3;
+    CeedScalar(*dirichlet_ceed)[num_comp];
+    CeedVectorGetArray(dirichlet_vector, CEED_MEM_HOST, (CeedScalar **)&dirichlet_ceed);
+
+    // set the boundary values
+    for (PetscInt i = 0; i < boundary.num_edges; ++i) {
+      dirichlet_ceed[i][0] = bc_values[num_comp * i];
+      dirichlet_ceed[i][1] = bc_values[num_comp * i + 1];
+      dirichlet_ceed[i][2] = bc_values[num_comp * i + 2];
```
Comment (bishtgautam) [inline src/rdydata.c:74]: This is pretty neat and I think we should implement a similar code for [RDySetWaterSource()](https://github.com/RDycore/RDycore/blob/93285b4303002a000f127e81310197c28f9771b0/src/rdydata.c#L121-L131)

**Comment by jeff-cohere** [comment]
Hmmm, it looks like we're not actually setting DIrichlet boundary values anywhere. I'll try to patch that in.

File: src/rdydata.c (hunk 1 above)
Comment (jeff-cohere) [reply src/rdydata.c:74]: Yeah, I was thinking the same thing. It makes our interface a bit smaller. I could do that in this PR, in fact, after I make sure we're actually testing the code that sets the Dirichlet BC values.

File: src/swe/swe_ceed.c [hunk 2]
```diff
@@ -263,6 +291,45 @@ PetscErrorCode CreateSWEFluxOperator(Ceed ceed, RDyMesh *mesh, int num_boundarie
...
+  CeedCompositeOperatorGetSubList(flux_op, &sub_ops);
+  CeedOperator boundary_flux_op = sub_ops[1 + boundary.index];
+
+  // fetch the field
+  CeedOperatorGetFieldByName(boundary_flux_op, "q_dirichlet", boundary_values);
+
+  PetscFunctionReturn(0);
+}
+
+PetscErrorCode SWEFluxOperatorSetDirichletBoundaryValues(CeedOperator flux_op, RDyBoundary boundary, PetscReal *boundary_values) {
```
Comment (bishtgautam) [inline src/swe/swe_ceed.c:309]: Should we also send `num_comp` and `num_edges` as input arguments so we know the size of `*boundary_values` AND an `assert` to check the size of  `*boundary_values`?

Comment (jeff-cohere) [reply src/swe/swe_ceed.c:309]: Sure. This is a matter of taste, but as you point out it may help us find errors and inconsistencies.

Comment (bishtgautam) [reply src/swe/swe_ceed.c:309]: Something similar to https://github.com/RDycore/RDycore/pull/94/files#diff-2380356150436250abd03a1522da01310c2e79d6104107b17cb4667e1d723fe4R41

**Review by bishtgautam — APPROVED** [review]

File: src/swe/swe_ceed.c (hunk 2 above)
Comment (bishtgautam) [reply src/swe/swe_ceed.c]:
```
PetscErrorCode SWEFluxOperatorSetDirichletBoundaryValues(CeedOperator flux_op, RDyBoundary boundary, PetscReal boundary_values[boundary.num_edges * ndof])
```

Comment (jeff-cohere) [reply src/swe/swe_ceed.c]: I think it's okay to handle these checks in `RDySetDirichletBoundaryValues` for now, but if we start to see these sorts of bugs crop up in our internal code, we can always add these parameters.

**Pull request merged**